src/main/scala/org/apache/sysml/api/dl/Caffe2DML.scala

```diff
@@ -184,7 +191,23 @@ class Caffe2DML(val sc: SparkContext, val solverParam:Caffe.SolverParameter,
   
   def getTrainAlgo():String = if(inputs.containsKey("$train_algo")) inputs.get("$train_algo") else "minibatch"
   def getTestAlgo():String = if(inputs.containsKey("$test_algo")) inputs.get("$test_algo") else "minibatch"
-    
+
+  def summary(sparkSession:org.apache.spark.sql.SparkSession):Unit = {
+    val header = Seq("Name", "Type", "Output", "Weight", "Bias", "Top", "Bottom")
+    val entries = net.getLayers.map(l => (l, net.getCaffeLayer(l))).map(l => {
+      val layer = l._2
+      (l._1, layer.param.getType, 
+          "(, " + layer.outputShape._1 + ", " + layer.outputShape._2 + ", " + layer.outputShape._3 + ")",
```

Comment:
The layer outputs are matrix, but logically Caffe2DML treats it as 4D and last dimensions are squashed.